### ISSUE TITLE
About zju-mocap Camera parameters question

### ISSUE THREAD

**Issue by Musosuke** [skhu101/GauHuman#53]
Thanks for your teams this tool, I have some question about dataset input, 

1. I am using this tool to manipulate Gaussian avatar poses. Instead of using it for rasterization, I am using it to transform a 3D Gaussian avatar and save the resulting file. Are there any specific considerations, such as scale or rotation, that I should be aware of after performing the transformation?
I noticed that, although the rotations for each vertex are applied into the gaussians, this does not fully resolve the jitter issue. My understanding is that applying LBS transform (rotation/position) to override the Gaussian rotations for Big Pose should produce correct results. However, I am still encountering problems, as shown when viewing the avatar in the WebGL splat viewer (see example below).

2. I am using the 4Ddress dataset for training. However, I noticed that the camera parameters only include K, R, and T,  and without distortion parameters. I am unsure whether the distortion values are implicitly padded as zero. Would this omission affect the visual quality of the trained character, potentially causing it to appear blurred?

I trained the model with missing  the distortion parameters from the 4Ddress dataset, and the evaluation scores and loss are approximately as follows:

[4ddress dataset] Inner_122, Take 2
[ITER 1200] Evaluating test #150: 
L1 0.007080945534010729 
PSNR 27.05699312845866 
SSIM 0.9552327613035838 
LPIPS 0.04671971440315247 [26/11 18:36:55]

[ITER 1200] Evaluating train #450: 
L1 0.0074998697157328325 
PSNR 27.03045878092448 
SSIM 0.9566293256812626 
LPIPS 0.04375596148272355 [26/11 18:37:30]

then, 
zju-mocap 377
[ITER 1200] Evaluating test #374: 
PSNR 32.238009 
SSIM 0.975874 
LPIPS 0.0189934578 [11/10 14:21:39]

Could you please clarify if the lack of distortion parameters might influence the results, and if so, how this can be addressed?

The following attachment is I did pose estimation to one frame from chinitaimei video, uses its pose to render my motion,
![image](https://github.com/user-attachments/assets/5322fc8d-04ea-4e9b-9d29-51da066cc5bd)

The following attachment is I did transform a training pose from 4ddress dataset , and the render result made by render.py
![image](https://github.com/user-attachments/assets/b58c16e1-4dba-4343-8b3d-f7c832589c25)

And I think probably another problem is that, I compared 4ddress dataset picture sequences, and zju-mocap dataset picture sequences,  I think zju-mocap did fully 360 deg rotation in any single view, 4ddress image sequence almost no rotation from lower body, will this condition makes the reconstruction/training quality issue from single view?

Any insights, tips, or recommendations would be greatly appreciated! Thank you very much!

**Comment by skhu101** [comment]
Hi, thanks for your interest in our work!

1. Your understanding is correct—we perform LBS transformations on 3D Gaussians.
2. If the distortion parameters are not included in the original dataset, it should be fine. In my experiments, I found that distortion parameters become important only if the original intrinsic parameters include distortion. For more details, you can refer to the getProjectionMatrix_refine function here: [link to GitHub](https://github.dev/skhu101/GauHuman/blob/732ec0d6783a29743f607cc0dce34ac08f57d0b0/utils/graphics_utils.py#L82).
3. Our current method appears to learn all available information from the provided monocular sequence. However, if the sequence does not include a full 360-degree rotation in any single view, the generated results may appear blurry due to the lack of information for those unseen views.

Hope this would be helpful to you.

**Comment by ZcsrenlongZ** [comment]
> Thanks for your teams this tool, I have some question about dataset input,
> 
>     1. I am using this tool to manipulate Gaussian avatar poses. Instead of using it for rasterization, I am using it to transform a 3D Gaussian avatar and save the resulting file. Are there any specific considerations, such as scale or rotation, that I should be aware of after performing the transformation?
>        I noticed that, although the rotations for each vertex are applied into the gaussians, this does not fully resolve the jitter issue. My understanding is that applying LBS transform (rotation/position) to override the Gaussian rotations for Big Pose should produce correct results. However, I am still encountering problems, as shown when viewing the avatar in the WebGL splat viewer (see example below).
> 
>     2. I am using the 4Ddress dataset for training. However, I noticed that the camera parameters only include K, R, and T,  and without distortion parameters. I am unsure whether the distortion values are implicitly padded as zero. Would this omission affect the visual quality of the trained character, potentially causing it to appear blurred?
> 
> 
> I trained the model with missing the distortion parameters from the 4Ddress dataset, and the evaluation scores and loss are approximately as follows:
> 
> [4ddress dataset] Inner_122, Take 2 [ITER 1200] Evaluating test #150: L1 0.007080945534010729 PSNR 27.05699312845866 SSIM 0.9552327613035838 LPIPS 0.04671971440315247 [26/11 18:36:55]
> 
> [ITER 1200] Evaluating train #450: L1 0.0074998697157328325 PSNR 27.03045878092448 SSIM 0.9566293256812626 LPIPS 0.04375596148272355 [26/11 18:37:30]
> 
> then, zju-mocap 377 [ITER 1200] Evaluating test #374: PSNR 32.238009 SSIM 0.975874 LPIPS 0.0189934578 [11/10 14:21:39]
> 
> Could you please clarify if the lack of distortion parameters might influence the results, and if so, how this can be addressed?
> 
> The following attachment is I did pose estimation to one frame from chinitaimei video, uses its pose to render my motion, ![image](https://github.com/user-attachments/assets/5322fc8d-04ea-4e9b-9d29-51da066cc5bd)
> 
> The following attachment is I did transform a training pose from 4ddress dataset , and the render result made by render.py ![image](https://github.com/user-attachments/assets/b58c16e1-4dba-4343-8b3d-f7c832589c25)
> 
> And I think probably another problem is that, I compared 4ddress dataset picture sequences, and zju-mocap dataset picture sequences, I think zju-mocap did fully 360 deg rotation in any single view, 4ddress image sequence almost no rotation from lower body, will this condition makes the reconstruction/training quality issue from single view?
> 
> Any insights, tips, or recommendations would be greatly appreciated! Thank you very much!

Hi! I am trying to employ GauHuman on 4DDress dataset. But I render black image. I guess it may due to the incorrect processing extrinsics. Could you help me to check if it is right? Many Thanks!

![Image](https://github.com/user-attachments/assets/81c73451-b3a3-4e8f-a5d2-d2a925296b8a)

**Comment by Musosuke** [comment]
your major problem is probably be:
1. zjumocap camera unit is probably 1mm, and 4ddress dataset use unit of translation is 1 meter 
![Image](https://github.com/user-attachments/assets/4b9bbc87-cc80-45f6-b5f1-43ffa852edb6)

2. you need to convert standard SMPL poses definition to neuralBody/zjumocap format，Rh[3] and poses[69] is not a behavior  just rename global_orient and poses
```
def trans_to_npyformat(data):
    from easymocap.bodymodel.smpl import SMPLModel
    global_orient = data['global_orient']
    dummy_orient = np.array([0, 0, 0])
    body_pose = data['body_pose']

    poses = np.concatenate([dummy_orient, body_pose])

    model_path = "SMPL_NEUTRAL.pkl"
    regressor_path = "J_regressor_body25.npy"
    """ do standard smpl format to zju-mocap convert here """
    converter = SMPLModel(model_path = model_path, regressor_path = regressor_path)

    poses = np.concatenate([global_orient, body_pose]).reshape(1, -1)
    convert_result = converter.convert_from_standard_smpl(
        {
            'poses': poses,
            'Th': data['transl'].reshape(1, -1), 
            'shapes': data['betas'].reshape(1, -1)
        }
        ) # input the poses, shapes, Th

    return convert_result
```

here is the explanation:
```
According to  zju-mocap format issue:
https://github.com/zju3dv/EasyMocap/issues/363

the maintainer mentioned: 
Please check this line: https://github.com/zju3dv/EasyMocap/blob/0fb7de5499baee32e67aedcd31047d424815b197/easymocap/bodymodel/smpl.py#L239

the funciton:
    def convert_from_standard_smpl(self, params):
        params = self.check_params(params)
        poses, shapes = params['poses'], params['shapes']
        Th = params['Th']
        vertices, joints, _, _ = lbs(shapes, poses, self.v_template,
                                self.shapedirs, self.posedirs,
                                self.J_regressor, self.parents,
                                self.weights, pose2rot=True, dtype=self.dtype, only_shape=True)
        # N x 3
        j0 = joints[:, 0, :]
        Rh = poses[:, :3].clone()
        # N x 3 x 3
        rot = batch_rodrigues(Rh)
        Tnew = Th + j0 - torch.einsum('bij,bj->bi', rot, j0)
        poses[:, :3] = 0
        res = dict(poses=poses.detach().cpu().numpy(),
            shapes=shapes.detach().cpu().numpy(),
            Rh=Rh.detach().cpu().numpy(),
            Th=Tnew.detach().cpu().numpy()
            )
        return res

To convert standard SMPL info into zju-mocap style,
need to do this:
https://github.com/zju3dv/EasyMocap/blob/master/doc/02_output.md#attention-for-smplsmpl-x-users

according the description, 'global_orient' to 'Rh' need to do following convertion step:
    We set the first 3 dimensions of poses to zero, 
    and add a new parameter Rh to represents the global oritentation, 
    the vertices of SMPL model V = RX(theta, beta) + T. 
    
    Please note that the paramter Rh is not equal to global_orient in SMPL-X model. 
    We take this representation because that changing paramters to new coordinate system in origin is difficult(see this link).

    In our representation, you can just use R'@(RX + T) + T' to convert the model, 
    and the new global rotaion and translation is simply written as R'@R and R'@T + T'
```
above content I think it would solve your current problem.


By the way, my reconstruction render result metrics is still with issues which metrics are worse over 5% than metric says in GauHuman paper, 
```
4D-DRESS00147_Inner
PSNR 29.103914455837675 SSIM 0.9522943512598674 LPIPS 0.05021028622570965 
```
If you found anything to adjust code more than I said which can be better to reconstruct in single or double view, please tell me too. 
my email: denghsian@gmail.com
